libcaf_core/caf/flow/op/merge.hpp

```diff
@@ -306,6 +308,7 @@ class merge_sub : public subscription::impl_base,
       else
         ++i;
     }
+    sub_.dispose();
```

Comment:
Calling `dispose()` might call `on_complete` or `on_error`, so we must always first reset `sub_` and then call `dispose()`. This is the general pattern we use:

```suggestion
    auto sub = std::move(sub_);
    sub.dispose();
```